packages/cli/src/runner/file-command-runner.ts

```diff
@@ -30,67 +33,83 @@ export class FileCommandRunner implements CommandRunner {
-            const jsonSummryReportName = this.reportDiskWriter.writeToDirectory(
-                scanArguments.output,
-                `ViolationCountByRuleMap`,
-                'json',
-                this.jsonSummaryReportGenerator.generateReport(this.summaryReportData),
-            );
-            console.log(`ViolationCountByRuleMap summary was saved in file ${jsonSummryReportName}`);
-        });
+        await promise;
+
+        await this.generateSummaryReport(scanArguments);
     }
 
-    private async scanURL(url: string): Promise<string> {
-        let axeResults: AxeScanResults;
+    private async generateSummaryReport(scanArguments: ScanArguments): Promise<void> {
+        console.log(this.consoleSummaryReportGenerator.generateReport(this.summaryReportData));
+
+        const jsonSummaryReportName = this.reportDiskWriter.writeToDirectory(
+            scanArguments.output,
+            `scan-summary`,
+            'json',
+            this.jsonSummaryReportGenerator.generateReport(this.summaryReportData),
+        );
+        console.log(`scan summary json was saved in file ${jsonSummaryReportName}`);
+
+        const htmlSummaryReportName = this.reportDiskWriter.writeToDirectory(
+            scanArguments.output,
+            `scan-summary`,
+            'html',
+            this.htmlSummaryReportGenerator.generateReport(this.summaryReportData),
+        );
+        console.log(`scan summary html was saved in file ${htmlSummaryReportName}`);
+    }
 
-        axeResults = await this.scanner.scan(url);
+    private async processUrl(url: string, scanArguments: ScanArguments): Promise<void> {
+        const axeResults = await this.scanner.scan(url);
 
-        this.processURLScanResult(axeResults);
+        if (isNil(axeResults.error)) {
+            const reportContent = this.reportGenerator.generateReport(axeResults);
+            const reportName = this.reportDiskWriter.writeToDirectory(scanArguments.output, url, 'html', reportContent);
 
-        return this.reportGenerator.generateReport(axeResults);
+            this.processURLScanResult(url, reportName, axeResults);
+        } else {
+            this.summaryReportData.unScannableUrls.push(url);
+        }
     }
 
-    private processURLScanResult(axeResults: AxeScanResults): void {
-        axeResults.results.violations?.forEach((violation) => {
-            this.summaryReportData.violationCountByRuleMap[violation.id] =
-                // tslint:disable-next-line: strict-boolean-expressions
-                (this.summaryReportData.violationCountByRuleMap[violation.id]
-                    ? this.summaryReportData.violationCountByRuleMap[violation.id]
-                    : 0) + violation.nodes.length;
-        });
+    private processURLScanResult(url: string, reportName: string, axeResults: AxeScanResults): void {
+        if (axeResults.results.violations?.length > 0) {
```

Comment:
this branch isn't 100% covered in unit test

Reply:
added tests. i had to refactor the tests to make sure that we validate the expected summary data. (this was not validated before even though coverage was 100%)